src/validation.cpp

```diff
@@ -762,7 +765,7 @@ static bool AcceptToMemoryPoolWorker(const CChainParams& chainparams, CTxMemPool
                 auto spend = TxInToZerocoinSpend(txin);
                 if (!spend)
                     return false;
-                if (spend->getVersion() < libzerocoin::CoinSpend::V3_SMALL_SOK)
+                if (spend->getVersion() < libzerocoin::CoinSpend::V3_SMALL_SOK || (fZCLimpMode && spend->getVersion() != libzerocoin::CoinSpend::V4_LIMP))
```

Comment:
These changes should not be showing up here. Need a proper rebase that does not grab already existing code and consider it as part of the changes made in this PR.